Orange/widgets/data/oweditdomain.py

```diff
@@ -229,11 +231,17 @@ def get_data(self):
     def is_same(self):
         """Is the current model state the same as the input.
         """
-        name = str(self.name_edit.text())
+        name = str(self.name_edit.text()).strip()
         labels = self.labels_model.get_dict()
-
-        return (self.var is not None and name == self.var.name and
-                labels == self.var.attributes)
+        if self.var_names is not None:
+            temp_var_names = list(self.var_names)
+            if self.var is not None and self.var.name in temp_var_names:
+                temp_var_names.remove(self.var.name)
+        else:
+            temp_var_names = []
+        return ((name in temp_var_names)
```

Comment:
This function checks whether it is *necessary* to change the domain. You shouldn't abuse it to check whether it's *legal* to change the domain. (See the general comment.)